web/src/test/java/com/microsoft/applicationinsights/web/utils/JettyTestServer.java

```diff
@@ -26,14 +26,41 @@
 import org.eclipse.jetty.servlet.FilterMapping;
 import org.eclipse.jetty.servlet.ServletContextHandler;
 
+import java.io.IOException;
+import java.net.ServerSocket;
+
 /**
  * Created by yonisha on 2/3/2015.
  */
 public class JettyTestServer {
     private Server server;
-    private int Min = 1050;
-    private int Max = 15000;
-    private int portNumber = Min + (int)(Math.random() * ((Max - Min) + 1));
+    private final int Min = 1050;
+    private final int Max = 15000;
+    private final int portNumber;
+
+    public JettyTestServer() {
+        // try 256 times for ports
+        int initialPortNumber = Min + (int)(Math.random() * ((Max - Min) + 1));
```

Comment:
A standard way / best practice in java to get random integer within the range is this:

```
import java.util.concurrent.ThreadLocalRandom;

// nextInt is normally exclusive of the top value,
// so add 1 to make it inclusive
int randomNum = ThreadLocalRandom.current().nextInt(min, max + 1);
```
Reference: https://stackoverflow.com/questions/363681/how-to-generate-random-integers-within-a-specific-range-in-java 

@littleaj it would be great if you can change this one as bonus!

Reply:
Since this is test code, so I'm not really worried about this.